WooCommerce/Classes/ViewRelated/Orders/OrdersViewModel.swift

```diff
@@ -124,6 +118,30 @@ final class OrdersViewModel {
         }
     }
 
+    private func createResultsControllerPredicate() -> NSPredicate {
+        let predicate: NSPredicate = {
+            let excludeSearchCache = NSPredicate(format: "exclusiveForSearch = false")
+            let excludeNonMatchingStatus = statusFilter.map { NSPredicate(format: "statusKey = %@", $0.slug) }
+
+            var predicates = [ excludeSearchCache, excludeNonMatchingStatus ].compactMap { $0 }
+            if !includesFutureOrders, let nextMidnight = Date().nextMidnight() {
+                // Exclude orders on and after midnight of today's date
+                let dateSubPredicate = NSPredicate(format: "dateCreated < %@", nextMidnight as NSDate)
+                predicates.append(dateSubPredicate)
+            }
+
+            return NSCompoundPredicate(andPredicateWithSubpredicates: predicates)
+        }()
+
+        guard let siteID = stores.sessionManager.defaultStoreID else {
+            assertionFailure("Trying to create results controller without a site ID")
+            return predicate
+        }
+
+        let siteIDPredicate = NSPredicate(format: "siteID = %lld", siteID)
+        return NSCompoundPredicate(andPredicateWithSubpredicates: [siteIDPredicate, predicate])
```

Comment:
After seeing this code duplicated, I wonder if we should extract the predicates creation into a helper to avoid duplication? 🤔

Reply:
since we are most likely deprecating iOS 12 support in the near future now that iOS 14 is gaining adoption, I think it's okay to keep the code in two places like the rest of Order list implementation. the iOS 12 usage is already low enough that we're ready to drop it, but we wanted to wait for Products M3 full release (release from this sprint!) so that users on iOS 12 get this feature.